dockers/docker-orchagent/supervisord.conf

```diff
@@ -26,6 +26,18 @@ stdout_logfile=syslog
 stderr_logfile=syslog
 dependent_startup=true
 
+[program:gearsyncd]
+command=/usr/bin/gearsyncd -p /usr/share/sonic/hwsku/gearbox_config.json
```

Comment:
what if the hwsku does not gearbox_config.json, will this break those platforms?

Reply:
The presence of this file in the hwsku folder indicates that gearbox should be enabled/configured. The gearbox_config.json should not be precent if gearbox is not supported.

Reply:
where is the condition? it seems this program will be running for other platforms even without gearbox present.

Reply:
@lguohan It starts to see if there is gearbox configuration in configDB, and if so writes it to REDIS for use by portsorch. If there is no configDB, it processes it by parsing the gearbox config file directly. Then it exits after signaling it is done. If there is no gearbox (determined by presence of a gearbox_config.json file), it just signals it is done, and exits (no need to look in configDB or parse the config file if device does not have a gearbox). 

We don't sit in a loop, we do the above, then exit, regardless if there is a gearbox or not.

https://github.com/Azure/SONiC/blob/b817a12fd89520d3fd26bbc5897487928e7f6de7/doc/gearbox/gearbox_mgr_design.md#312-gearsyncd-new